shell/server/installer.js

```diff
@@ -102,6 +128,28 @@ cancelDownload = function (packageId) {
   }
 }
 
+if (!Meteor.settings.replicaNumber) {
+  installers = {};
+
+  Meteor.startup(function () {
+    // Restart any deletions that were killed while in-progress.
+    Packages.find({status: "delete"}).forEach(function (pkg) {
+      deletePackageInternal(pkg._id);
+    });
+
+    // Watch for new installation requests and fulfill them.
+    Packages.find({status: {$in: ["download", "verify", "unpack", "analyze"]}}).observe({
+      added: startInstallInternal,
+      removed: cancelDownloadInternal
+    });
+
+    // Watch for new cleanup requests and fulfill them.
+    Packages.find({status: "ready", shouldCleanup: true}).observe({
+      added: deletePackageInternal
```

Comment:
Here's a case I'm worried about: What happens if `deleteUnusedPackages(appId)` gets called on an app that has a package-install in progress? It looks like `shouldCleanup` will get set to `true` and then once the package is "ready" then `deletePackageInternal()` will be called, perhaps before `addUserActions()` has had a chance to run!

Possible alternative strategy: don't include `status` in the query here, and then in `AppInstaller.prototype.done()` clear then value of `shouldCleanup`. Then if a delete happens during an install, the delete gets ignored, but can be retried later.